dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/HostBundlerHandlerTest.java

```diff
@@ -0,0 +1,156 @@
+package com.dotcms.enterprise.publishing.remote;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.LicenseTestUtil;
+import com.dotcms.enterprise.publishing.remote.bundler.HostBundler;
+import com.dotcms.enterprise.publishing.remote.handler.HostHandler;
+import com.dotcms.publisher.pusher.PushPublisherConfig;
+import com.dotcms.publishing.BundlerStatus;
+import com.dotcms.publishing.DotBundleException;
+import com.dotcms.publishing.PublisherConfig;
+import com.dotcms.publishing.PublisherConfig.Operation;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.db.HibernateUtil;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.util.Config;
+import com.liferay.portal.model.User;
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class HostBundlerHandlerTest extends IntegrationTestBase {
+
+    private static User user;
+    private static HostAPI hostAPI;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+        LicenseTestUtil.getLicense();
+
+        hostAPI = APILocator.getHostAPI();
+        user = APILocator.getUserAPI().getSystemUser();
+    }
+
+    @Test
+    public void testBundlerHandler_Unpublish_Success() throws Exception {
+        final BundlerStatus status;
+        final String assetRealPath = Config.getStringProperty("ASSET_REAL_PATH", "test-resources");
+        final File tempDir = new File(assetRealPath + "/bundles/" + System.currentTimeMillis());
+        final HostBundler hostBundler;
+        final PushPublisherConfig config;
+        final Set<String> contentSet;
+
+        try {
+
+            contentSet = new HashSet();
+            Host host = new Host();
+            hostBundler = new HostBundler();
+            status = new BundlerStatus(HostBundler.class.getName());
+
+            //Creating host
+            host.setHostname("hostUnpublish" + System.currentTimeMillis() + ".dotcms.com");
+            host.setDefault(false);
+
+            HibernateUtil.startTransaction();
+            host = APILocator.getHostAPI().save(host, user, false);
```

Comment:
when is this host deleted?

Reply:
With the unpublish operation that is executed after in the handler.